marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/xquery-lib.xqy

```diff
@@ -61,8 +62,34 @@ declare function data-hub-map-to-canonical(
     else map:map()=>map:with("template", $target-entity-name)
   let $results :=
     xdmp:xslt-invoke($mapping-uri||".xslt", $input, $user-params, $parms)
+  let $x := xdmp:log($results)
   return
     if ($format="xml")
     then inst:canonical-xml($results)
-    else inst:canonical-json($results)
+    else(
+      let $element := $results/element()
+      return
+      (: This change is necessitated by DHFPROD-6219. In case the entity doesn't have strucutred properties, it
+         returns empty json object. If it has structured properties, the empty structured properties are removed.
+         inst:canonical-json() method introduces the '$ref' .  :)
+      if (empty($element/*)) then
+        document{json:object()=>map:with(string(fn:node-name($element)), json:object())}
+      else
+        inst:canonical-json(xquery-lib:clean-xml($element))
+    )
+};
+
+
+declare function clean-xml($element as item()*) as document-node() {
+  document{
+    element {fn:node-name($element)} {
+      $element/namespace::node(),
+      $element/@*,
+      for $child in $element/element()
+      return
```

Comment:
Since the if condition returns an empty sequence, this could be changed to a where clause with a not().

Reply:
Added a where clause with a not()